include/frame.h

```diff
@@ -17,20 +17,27 @@
 #include <vector>
 
 #include "refcount.h"
+#include "streamable_flatbuffer.h"
+#include "../fbs/torchcraft_generated.h"
 
 #ifdef _MSC_VER
 typedef unsigned int uint32_t;
 typedef unsigned short uint16_t;
 typedef int int32_t;
 #endif
 
-// TODO Check types !
-
 namespace torchcraft {
 namespace replayer {
+
+class Frame;
+class FrameDiff;
+std::ostream& operator<<(std::ostream& out, const Frame& o);
+std::istream& operator>>(std::istream& in, Frame& o);
+std::ostream& operator<<(std::ostream& out, const FrameDiff& o);
+std::istream& operator>>(std::istream& in, FrameDiff& o);
```

Comment:
Can you maybe add read/write methods for this as well? I'm not a fan of overloading the stream operators.

Reply:
The implementations are now basically thin wrappers around read/write methods. See addToFlatBufferBuilder() and readFromFlatBufferTable()

I'm not a fan either; they're basically class methods floating in space, it's really hard to grep for usages, and any problems with the overload signatures produce horrific compiler errors. 

Ideally we wouldn't be using streams until the last minute anyhow, since they introduce an additional copy.

Reply:
Ok, sounds good!